### DISCUSSION TITLE
Multi shop

### DISCUSSION THREAD

**Discussion by mc0657** [lunarphp/lunar#1440] [Feature Requests]
Hello,

I've been exploring Lunar with great interest. Disgusted by Prestashop, I decided two years ago to develop a website from scratch, which has been in production for 9 months now. I am now looking to elevate it to the next level. Interestingly, my code has a lot of similarities with Lunar's and employs the same technologies (TALL, Spatie packages, Scout...), so I could consider migrating to it quite easily and focus on more high level features rather than on the ecommerce core.

However, one aspect of the project that surprises me is the absence of a Shop concept in the core. How are we supposed to manage multi-shop scenarios? For instance, how would a product that has one price in shop1 and another in shop2, or has a discount of X€ in shop1 and Y€ in shop2, be handled?

Did I miss something, or has the multi-shop aspect not been integrated? Is it a feature that is being considered?

**Comment by glennjacobs** [comment]
We have (Sales) Channels which work in a similar way, however, we don't currently support different pricing per-channel.

**Reply by mc0657** [reply]
Yes, that's what I've indeed noticed. But beyond pricing, multi-shop involves many other implications, such as:
- Payment methods differ from one shop to another
- Shipping terms and authorized delivery areas vary between shops
- Customer groups (and their related pricing condition) may vary between shops
- It is strongly recommended that media have different URLs, using the distinct domain names of each shop
- Etc.

Would you be open to contributions for implementing the multi-shop concept? Or do you consider that this falls outside the core's role and should be addressed through extensions?

**Reply by glennjacobs** [reply]
I would probably suggest that anything like this would want to be done on the Filament admin panel which is currently a work in progress. Whether its part of the core or an add-on I'm not sure at this stage.

Multi-Store has come up before, but I think it needs quite a bit of consideration before anything is implemented.

One of our goals is to try keep the system as simple as possible, and this naturally could add a lot of complexity.
